src/coreclr/nativeaot/System.Private.StackTraceMetadata/src/Internal/StackTraceMetadata/StackTraceMetadata.cs

```diff
@@ -290,12 +314,43 @@ public string GetMethodNameFromRvaIfAvailable(int rva)
                 }
 
                 StackTraceData data = _stacktraceDatas[index];
+
+                isStackTraceHidden = data.IsHidden;
+
+                if (data.OwningType.IsNull(_metadataReader))
+                {
+                    Debug.Assert(data.Name.IsNull(_metadataReader) && data.Signature.IsNull(_metadataReader));
+                    Debug.Assert(isStackTraceHidden);
+                    return null;
+                }
+
                 return MethodNameFormatter.FormatMethodName(_metadataReader, data.OwningType, data.Name, data.Signature, data.GenericArguments);
             }
 
             private struct StackTraceData : IComparable<StackTraceData>
             {
-                public int Rva { get; init; }
+                private const int IsHiddenFlag = 0x1;
```

Comment:
Stealing the lowest bit like this tends to be a problem on Arm32, but we can deal with it if we ever get there.

Reply:
We can just switch this to second highest bit - we already align everything at 4 bytes minimum because of ARM and fat function pointers. The assert should help guarding against this ever breaking the invariant.

```suggestion
                private const int IsHiddenFlag = 0x2;
```